src/datetime.zig

```diff
@@ -313,30 +313,30 @@ pub const Date = struct {
     // Create a date from the number of seconds since 1 Jan 1970
     pub fn fromSeconds(seconds: f64) Date {
         const r = math.modf(seconds);
-        const timestamp = @intFromFloat(i64, r.ipart); // Seconds
+        const timestamp: i64 = @intFromFloat(r.ipart); // Seconds
         const days = @divFloor(timestamp, time.s_per_day) + @as(i64, EPOCH);
         assert(days >= 0 and days <= MAX_ORDINAL);
-        return Date.fromOrdinal(@intCast(u32, days));
+        return Date.fromOrdinal(@intCast(days));
     }
 
     // Return the number of seconds since 1 Jan 1970
     pub fn toSeconds(self: Date) f64 {
-        const days = @intCast(i64, self.toOrdinal()) - @as(i64, EPOCH);
-        return @floatFromInt(f64, days * time.s_per_day);
+        const days = @as(i64, @intCast(self.toOrdinal())) - @as(i64, EPOCH);
```

Comment:
Is this `@as` needed?

```suggestion
        const days = @intCast(self.toOrdinal()) - @as(i64, EPOCH);
```

Reply:
![2023-06-27_16-06](https://github.com/frmdstryr/zig-datetime/assets/9482395/d297bb5c-5874-4e41-ba8b-42cf7681d18a)



It seems necessary. I will test all your suggestions locally. Please wait a moment.

Reply:
Hmm. Though the rhs would make sure that the result location was `i64` :shrug: